src/pybind/mgr/dashboard/services/ceph_service.py

```diff
@@ -114,14 +106,20 @@ def get_pool_list_with_stats(cls, application=None):
 
             def get_rate(series):
                 if len(series) >= 2:
-                    return differentiate(*list(series)[-2:])
+                    return derivative(*list(series)[-2:], True)
                 return 0
 
             for stat_name, stat_series in stats.items():
+                # Using the absolute value for 'rate' and 'rates' is a temporary workaround for
+                # the pg decrease issue. A decrease will reduce the maximum amount in all time
```

Comment:
Could you please add here the URL to the issue? I couldn't find it.

I do think we must not sweep issues under the rug. If we are facing duplicate samples, mismatching samples for the same timestamp, or counters that are expected to behave monotonically but somehow decrease, we shouldn't make it up. This is the proof that Ceph-Dashboard is doing a great job exposing issues.

So I still think we should remove the `abs()` (now it can be optionally enabled... but it's enabled most of the times!) and the ` deriv = (p2[1] - p1[1]) / (time_diff if time_diff != 0 else 1)` should be just `deriv = (p2[1] - p1[1]) / (p2[0] - p1[0])`.